### PR TITLE
Repair sign of projection inverse retraction for Stiefel

### PR DESCRIPTION
**Pull request by sethaxen**

### CODE REVIEW

**Review by mateuszbaran — APPROVED** [review]
Sorry, that looks like an issue with my suggestion here: https://github.com/JuliaManifolds/Manifolds.jl/pull/535#discussion_r991647890 -- thank you for fixing it.

**Comment by mateuszbaran** [comment]
Yes, inner product with a zero vector is always equal to 0. Good catch!

**Comment by sethaxen** [comment]
Should this just check the inner product of the two vectors directly, or just check the norm of the difference?

**Comment by mateuszbaran** [comment]
I guess it was supposed to be `isapprox(norm(M, p, X - Y), 0; kwargs...)`. It might need tweaking the default `atol` though.

**Comment by sethaxen** [comment]
> It might need tweaking the default `atol` though.

We don't do that anywhere else though. So this would probably be more stable as a default: `isapprox(inner(M, p, X, Y), norm(M, p, X) * norm(M, p, Y); kwargs...)`

**Comment by mateuszbaran** [comment]
Good question. Maybe it makes some minor difference for the canonical metric but a simple `isapprox` would probably also be fine. @kellertuer what do you think?

**Comment by kellertuer** [comment]
It depends a little bit, which functions are implemented I think. I am in favour of the `-` version, though for that your tangent vectors have to have a `-` .

File: src/manifolds/Grassmann.jl [hunk 1]
```diff
@@ -99,10 +99,12 @@ injectivity_radius(::Grassmann, p) = π / 2
 injectivity_radius(::Grassmann, ::AbstractRetractionMethod) = π / 2
 injectivity_radius(::Grassmann, p, ::AbstractRetractionMethod) = π / 2
 
-function Base.isapprox(M::Grassmann, p, X, Y; kwargs...)
-    return isapprox(sqrt(inner(M, p, zero_vector(M, p), X - Y)), 0; kwargs...)
+function Base.isapprox(M::Grassmann, p, X, Y; atol=sqrt(max_eps(X, Y)), kwargs...)
```
Comment (mateuszbaran) [inline src/manifolds/Grassmann.jl:102]: `GeneralizedGrassmann` has the same issue, it would be worth fixing as well.

Comment (kellertuer) [reply src/manifolds/Grassmann.jl:102]: Yes, that would be great but the problem should be _exactly_ the same fix :)

Comment (sethaxen) [reply src/manifolds/Grassmann.jl:102]: Already fixed.

**Review by mateuszbaran — APPROVED** [review]

**Comment by sethaxen** [comment]
I had to increase a few tolerances (and use the tolerances in a few places they weren't being used in `test_manifold`), but now all tests for these manifolds should pass, save for one: `vector transport method DifferentiatedRetractionVectorTransport{QRRetraction}(QRRetraction())`. This one seems to fail regardless of the tolerance. Could one of you look into this?

**Comment by mateuszbaran** [comment]
I will try to find some time to look at it in the evening.

**Comment by sethaxen** [comment]
Thanks! Seems to have done the trick!

**Comment by kellertuer** [comment]
This looks great! Since we (for now) do that specific qr in two places (and I'll have to check whether GeneralisedStiefel is affected as well) we could also introduce a function for that in our utils?

**Comment by sethaxen** [comment]
Perhaps (it should be called `qr_unique`), but let's do that in another PR.

**Comment by mateuszbaran** [comment]
Yes, let's not delay this set of bugfixes further than necessary. We can work further on these QR improvements for Stiefel in another PR.

**Review by mateuszbaran — APPROVED** [review]

**Pull request merged**